src/components/__tests__/CustomDropdown.spec.tsx

```diff
@@ -124,20 +124,104 @@ describe('Custom dropdown component', () => {
                     status: ValueStatus.Available,
                     value: null,
                     setValue: contextObjIdSetValue
-                },
+                }
             });
 
-            const clearButton = component.container.querySelector('div.test__clear-indicator');
+            const clearButton = component.container.querySelector("div.test__clear-indicator");
             fireEvent.mouseDown(clearButton, { button: 0 });
-        })
+        });
 
-        it('- should show the placeholder', () => {
-            const placeholder = component.container.querySelector('div.test__value-container > div.test__placeholder');
+        it("- should show the placeholder", () => {
+            const placeholder = component.container.querySelector("div.test__value-container > div.test__placeholder");
             expect(placeholder).not.toBeNull();
         });
 
-        it('- should call clearValue action', () => {
+        it("- should call clearValue action", () => {
             expect(clearValue).toHaveBeenCalledTimes(1);
         });
     });
-});
\ No newline at end of file
+
+    describe("New entry is created", () => {
+        let component;
+        const createValue = jest.fn();
+        const contextObjLabelSetValue = jest.fn();
+        const contextObjIdSetValue = jest.fn();
+
+        beforeEach(() => {
+            component = renderComponent({
+                createValue: {
+                    canExecute: true,
+                    execute: createValue
+                },
+                contextObjLabel: {
+                    status: ValueStatus.Available,
+                    value: null,
+                    setValue: contextObjLabelSetValue
+                },
+                contextObjId: {
+                    status: ValueStatus.Available,
+                    value: null,
+                    setValue: contextObjIdSetValue
+                }
+            });
+
+            const inputValue = component.container.querySelector("div.test__input > input");
+            fireEvent.change(inputValue, { target: { value: "This is a new value" } });
+        });
+
+        it("- should show option to create value", () => {
+            expect(component.container.querySelector("div.test__menu-list > div.test__option")!.textContent).toBe(
+                'Create "This is a new value"'
+            );
+        });
+
+        it("- should create a value", () => {
+            const createOption = component.container.querySelector("div.test__menu-list > div.test__option");
+
+            fireEvent.keyDown(createOption, { key: "Enter", code: "Enter" });
+            expect(createValue).toHaveBeenCalledTimes(1);
+        });
```

Comment:
```suggestion
        describe("and Enter key down", () => {
            beforeEach(() => {
                const createOption = component.container.querySelector("div.test__menu-list > div.test__option");
                fireEvent.keyDown(createOption, { key: "Enter", code: "Enter" });
            });
            
            it("- should create a value", () => {                
                expect(createValue).toHaveBeenCalledTimes(1);
            });
        })
```
The reason is that this will affect any other tests added into top level `describe`.